packages/app/src/app/pages/Pro/index-v2.tsx

```diff
@@ -0,0 +1,342 @@
+} from '@codesandbox/components';
+import css from '@styled-system/css';
+import { useOvermind } from 'app/overmind';
+import { Navigation } from 'app/pages/common/Navigation';
+import { TeamAvatar } from 'app/components/TeamAvatar';
+import { NewTeam } from 'app/pages/common/NewTeam';
+import { TeamMemberAuthorization } from 'app/graphql/types';
+
+export const ProPage: React.FC = () => (
+  <ThemeProvider>
+    <Helmet>
+      <title>Pro - CodeSandbox</title>
+    </Helmet>
+    <Stack
+      direction="vertical"
+      css={css({
+        backgroundColor: 'grays.900',
+        color: '#fff',
+        width: ' 100vw',
+        minHeight: '100vh',
+      })}
+    >
+      <Navigation title="CodeSandbox Pro" />
+
+      <Switch>
+        <Route path={`/pro/create-workspace`}>
+          <NewTeam redirectTo="/pro?v=2" />;
+        </Route>
+        <Route path={`/pro`}>
+          <Upgrade />
+        </Route>
+      </Switch>
+    </Stack>
+  </ThemeProvider>
+);
+
+
+const prettyPermissions = {
+  ADMIN: 'Admin',
+  WRITE: 'Editor',
+  READ: 'Viewer',
+};
+
+const Upgrade = () => {
+  const {
+    state: { personalWorkspaceId, user, activeTeam, dashboard },
+    actions: { setActiveTeam },
+  } = useOvermind();
+
+  const location = useLocation();
+  const history = useHistory();
+  const workspaceId = new URLSearchParams(location.search).get('workspace');
+
+  React.useEffect(() => {
+    // set workspace in url when coming from places without workspace context
+    if (activeTeam && !workspaceId) {
+      history.replace({
+        pathname: location.pathname,
+        search: '?v=2&workspace=' + activeTeam,
+      });
+    } else if (workspaceId !== activeTeam) {
+      setActiveTeam({ id: workspaceId });
+    }
+  }, [workspaceId, activeTeam, history, location, setActiveTeam]);
+
+  if (!activeTeam || !dashboard.teams.length) return null;
+
+  const personalWorkspace = dashboard.teams.find(
+    t => t.id === personalWorkspaceId
+  )!;
+
+  const getUserAuthorization = workspace => {
+    const userAuthorization = workspace.userAuthorizations.find(
+      authorization => authorization.userId === user.id
+    ).authorization;
+    return userAuthorization;
+  };
+
+  const workspaces = [
+    personalWorkspace,
+    ...sortBy(
+      dashboard.teams.filter(t => t.id !== personalWorkspaceId),
+      [getUserAuthorization, 'name']
+    ),
+  ];
+
+  const activeWorkspace = workspaces.find(
+    workspace => workspace.id === activeTeam
+  );
+
+  const isPersonalWorkspace = personalWorkspaceId === activeTeam;
+  const numberOfViewers = activeWorkspace.userAuthorizations.filter(
+    ({ authorization }) => authorization === TeamMemberAuthorization.Read
+  ).length;
+  const numberOfEditors = activeWorkspace.userAuthorizations.filter(
+    ({ authorization }) => authorization !== TeamMemberAuthorization.Read
+  ).length;
+
+  const isPersonalPro = isPersonalWorkspace || user.subscription;
+  const isTeamPro = activeWorkspace.joinedPilotAt;
```

Comment:
Will this always exist?

Reply:
this will become the following until current folks are migrated over to new payment system

```js
const isTeamPro = activeWorkspace.joinedPilotAt || activeWorkspace.subscription
```